src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -246,8 +343,25 @@ const UsersFragment = ({
         <Text style={styles.noDataFoundText}>No users found</Text>
       </View>
     );
-
-  return <View>{renderUsersItem()}</View>;
+  return (
+    <View>
+      {loadingDM && (
+        <View
+          style={{
+            position: 'absolute',
+            zIndex: 1,
+            width: '100%',
+            height: '100%',
+            justifyContent: 'center',
+            alignContent: 'center',
+            alignSelf: 'center'
+          }}>
+          <ActivityIndicator size="large" />
+        </View>
+      )}
+      {renderUsersItem()}
+    </View>
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The loading state is being handled with a conditional rendering of an `ActivityIndicator` component. This could be abstracted into a separate reusable component for better maintainability and reusability.

```diff
- {loadingDM && (
-   <View
-     style={{
-       position: 'absolute',
-       zIndex: 1,
-       width: '100%',
-       height: '100%',
-       justifyContent: 'center',
-       alignContent: 'center',
-       alignSelf: 'center'
-     }}>
-     <ActivityIndicator size="large" />
-   </View>
- )}
+ <LoadingOverlay isLoading={loadingDM} />
```

In the above code, `LoadingOverlay` is a new component that encapsulates the loading state UI.


<!-- This is an auto-generated comment by OSS CodeRabbit -->